driver-core/src/main/com/mongodb/internal/async/client/AsyncMongoClientImpl.java

```diff
@@ -67,7 +67,7 @@ private AsyncMongoClientImpl(final MongoClientSettings settings, final Cluster c
                                  @Nullable final Closeable externalResourceCloser) {
         this.settings = notNull("settings", settings);
         this.cluster = notNull("cluster", cluster);
-        this.serverSessionPool = new ServerSessionPool(cluster);
+        this.serverSessionPool = new ServerSessionPool(cluster, settings.getServerApi().orElse(null));
```

Comment:
Is an optional better?

Reply:
IntelliJ actually complains when you specify an Optional as a parameter:

> Optional was designed to provide a limited mechanism for library method return types where there needed to be a clear way to represent "no result".

Reply:
Sorry I was really asking if using an optional in the settings is actually better / beneficial

Reply:
Right.  I don't think it is, and am removing it as per your suggestion

Reply:
Done